src/relay/op/contrib/ethosu/pooling.cc

```diff
@@ -46,14 +46,27 @@ bool EthosuPoolingRel(const Array<Type>& types, int num_inputs, const Attrs& att
 
   const String operator_name = "ethosu_pooling";
 
-  if (param->pooling_type != "AVG" && param->pooling_type != "MAX") {
+  if (param->pooling_type != "AVG" && param->pooling_type != "MAX" &&
+      param->pooling_type != "SUM") {
     reporter->GetDiagCtx().EmitFatal(Diagnostic::Error(reporter->GetSpan())
                                      << "Invalid operator: expected " << operator_name
-                                     << " type 'AVG' or 'MAX' but was " << param->pooling_type);
+                                     << " type 'AVG', 'MAX', or 'SUM' but was "
+                                     << param->pooling_type);
     return false;
   }
 
-  CheckDataType(reporter, ifm->dtype, {DataType::UInt(8), DataType::Int(8)}, operator_name, "ifm",
+  auto max_avg_pooling_ifm_dtypes = {DataType::UInt(8), DataType::Int(8)};
+  auto sum_pooling_ifm_dtypes = {DataType::UInt(8), DataType::Int(8), DataType::Int(16),
```

Comment:
Nit: prefer types over `auto`

Reply:
C++ code style says:

> Favor concrete type declaration over auto as long as it is short.

How short should it be?
Is it ok to use
`std::initializer_list<DataType>`
or better 
```
using DTypes = std::initializer_list<DataType>;
DTypes sum_pooling_ifm_dtypes
```
?

Reply:
I think it's okay to use `std::initializer_list<DataType>` in this instance

Reply:
Done.